### PR TITLE
Add deletionTimestamp support for core objects to API

### PR DESCRIPTION
**Pull request by floreks**
**What does this PR do / Why do we need it**:
Some of the core resources did not expose the `deletionTimestamp` as part of the `ObjectMeta` or did not have the `ObjectMeta` embedded at all. We need in the UI to be able to properly show the state of the resource and if it is being deleted.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -494,6 +494,13 @@ func ConvertInternalUserToExternal(internalUser *kubermaticv1.User, includeSetti
 			ID:                internalUser.Name,
 			Name:              internalUser.Spec.Name,
 			CreationTimestamp: NewTime(internalUser.CreationTimestamp.Time),
+			DeletionTimestamp: func() *Time {
+				if internalUser.DeletionTimestamp != nil {
+					deletionTimestamp := NewTime(internalUser.DeletionTimestamp.Time)
+					return &deletionTimestamp
+				}
+				return nil
+			}(),
```
Comment (xrstf) [inline pkg/api/v1/types.go:503]: You could expose this as a global function and then use it in all the handlers to convert timestamps. :)

**Review by xrstf — COMMENTED** [review]
A comment

File: pkg/api/v1/types.go (hunk 1 above)
Comment (floreks) [reply pkg/api/v1/types.go:503]: I am sure someone will eventually refactor this :)

**Pull request merged**